term/src/terminal.rs

```diff
@@ -51,6 +51,8 @@ pub enum Alert {
     TabTitleChanged(Option<String>),
     /// When the color palette has been updated
     PaletteChanged,
+    /// When the mouse cursor shape has been updated
+    MouseCursorShapeChanged,
```

Comment:
Let's pass the name along with this, like with the title changed event above.
That will make it easier to do a half-baked mux client implementation (see 44866cc137e336d8c00b23f5e0d7d1f03983b591)

```suggestion
    MouseCursorShapeChanged(Option<String>)
```

Reply:
Note that by changing the `Alert` enum, you're changing the binary signature of types on which the mux protocol depends. You'll need to bump the CODEC_VERSION:

https://github.com/wezterm/wezterm/blob/c06cc87d9c50b6f45a03a0c9bfe24381cbb194c5/codec/src/lib.rs#L441-L444